File: backend/app/schemas/backup.py
from datetime import datetime

from .base import BaseSchema
from pydantic import EmailStr


# -------------------------------------------------------------------
# Request
# -------------------------------------------------------------------
class BackupEmailRequest(BaseSchema):
    email: EmailStr

# -------------------------------------------------------------------
# Reusable
# -------------------------------------------------------------------
class BackupInfoOut(BaseSchema):
    last_backup_at: datetime
    last_backup_file: str
    last_backup_size: int
    type: str


class BackupFileOut(BaseSchema):
    filename: str
    created_at: datetime
    size: int
    type: str

# -------------------------------------------------------------------
# Status
# -------------------------------------------------------------------
class BackupStatusOut(BaseSchema):
    last_backup_at: datetime | None
    last_backup_file: str | None
    last_backup_size: int | None
    last_backup_source: str | None

    email_enabled: bool
    email_from: str | None

    last_automatic_backup_at: datetime | None
    next_automatic_backup_at: datetime | None

    last_backup_result: str | None

# -------------------------------------------------------------------
# Responses
# -------------------------------------------------------------------
class BackupEmailOut(BackupInfoOut):
    sent_to: EmailStr


class BackupDeleteOut(BaseSchema):
    success: bool


class BackupRestoreOut(BaseSchema):
    success: bool
    restart_required: bool

File: frontend/src/types/backup.ts
/**
 * Interface de backups
 */

export const BackupType = {
  MANUAL: "manual",
  DAILY: "daily",
  WEEKLY: "weekly",
  MONTHLY: "monthly",
  BEFORE_RESTORE: "before_restore"
} as const

export type BackupType =
  typeof BackupType[keyof typeof BackupType]


export interface BackupStatus {
  last_backup_at: string | null
  last_backup_file: string | null
  last_backup_size: number | null
  last_backup_source: BackupType | null

  email_enabled: boolean
  email_from: string | null

  last_automatic_backup_at: string | null
  next_automatic_backup_at: string | null
  last_backup_result: string | null
}


export interface BackupFile {
  filename: string
  created_at: string
  size: number
  type: BackupType
}


export interface BackupRestoreResponse {
  restart_required: boolean
}